src/utils/jellyfin-apiclient/getItems.ts

```diff
@@ -0,0 +1,61 @@
+import type {BaseItemDtoQueryResult} from '@jellyfin/sdk/lib/generated-client';
+import {ApiClient} from 'jellyfin-apiclient';
+
+const idsPerItemRequestLimit = 25;
+
+function getItemsSplit(apiClient: ApiClient, userId: string, options: any) {
```

Comment:
Instead of using `any`, adding a type in this file that covers the usage here should be preferred. (The end goal of course would be to use the TS SDK where everything is already typed, but it's not compatible here unfortunately)

Reply:
Is there documentation somewhere on what variables can be defined in options? Because how options looks like heavily depends on the caller (it varies a lot) and is often built through many function calls. I don't think I have a good enough understanding of the variable and how the object could look like. This is why I use this `optionsTemplate` and only change the `Ids` field.

Reply:
Actually I just looked it up in the generated docs of the typescript sdk and if I'm not mistaken it is also `any` there... (`docs/classes/generated_client.ItemsApi.md`)

Reply:
Here is what the SDK uses: https://typescript-sdk.jellyfin.org/interfaces/generated_client.ItemsApiGetItemsRequest.html (note the capitalization is camelCase instead of PascalCase which is why I said it wouldn't be compatible)

Since this is just a build time check you should only have to define the properties used in this file though...

```ts
interface GetItemsRequest {
    Ids?: string
    Limit?: number
}
```

Unless you would _want_ to add the whole thing to the (apiclient type definitions file)[https://github.com/jellyfin/jellyfin-web/blob/master/src/apiclient.d.ts].

Reply:
I see! I could only implement the necessary parts as an interface!
I'm new to typescript so I did not think of that. Thanks!

I implemented this now and am looking forward to your feedback!

Reply:
Ooops, yes. Not it's pushed ^^